### ISSUE TITLE
Allow clicking links on whitelisted sites

### ISSUE THREAD

**Issue by sean-public** [aniketpanjwani/chomper#10]
When in whitelisting mode, it should also whitelist domains linked by whitelisted sites so I could click links on that page (i.e. following a citation on a SO answer) and load resources linked on other domains (i.e. static files on a CDN).

This would involve parsing the response when it's passing through the proxy and pulling domains referenced in `link`/`a`/etc tags and adding the domains to the in-memory whitelist.

**Comment by miheerdew** [comment]
Does that mean I can go to any webpage if I whitelist google.com?

**Comment by sean-public** [comment]
Yes, exactly.

It also means that all the images will load on a given page, all the JS and CSS served from a CDN will load, etc.

At first I was thinking it might be behind a global flag or list-specific one, but now I'm starting to think that you may want more granular control at the domain level, even. For example, you might have a whitelist mode, blacklist mode, and then a whitelist-with-follow mode or similar. Then at the domain level you can set a flag for follow/nofollow.

**Comment by miheerdew** [comment]
If you just want to allow JS/CSS that can be done by allowing regular expressions in whitelists for those types of files. If you want to allow following links then maybe set a max_depth? In any case we should keep the [KISS](https://en.wikipedia.org/wiki/KISS_principle) principle in mind.

**Comment by sean-public** [comment]
I'm sure you'll figure something out that gives a good browsing experience. I don't think it will involve depth settings and regex, however.

**Comment by aniketpanjwani** [comment]
I agree that this would have to be done at the domain level to be useful.

However, I think that this feature is fairly complicated to develop relative to the extent to which it will drive greater usage of Chomper and improve usage of existing users. There are some more immediately pressing features and small bugs to fix which have a larger "increased utility gain to development time" ratio. I'm going to make explicit release plans within the next week in the Projects section.
